src/core/structure.js

```diff
@@ -826,13 +821,7 @@ p5.prototype.push = function() {
  * </div>
  */
 p5.prototype.pop = function() {
-  const style = this._styles.pop();
-  if (style) {
-    this._renderer.pop(style.renderer);
-    Object.assign(this, style.props);
-  } else {
-    console.warn('pop() was called without matching push()');
-  }
+  this._renderer.pop();
```

Comment:
do we still get a warning on `pop()` was called without matching `push()`?

Reply:
The `p5` instance does not have knowledge about whether any call to `pop()` is valid or not as the states are handled by the renderer. The warning message if needed can be added to `p5.Renderer` base class but I'm not entirely sure if it is useful.

The feature is also not complete (to be completed in #7270), this PR just to lay groundwork for converting the WebGL module to use the new module syntax.

Reply:
> ThYe `p5` instance does not have knowledge about whether any call to `pop()` is valid or not as the states are handled by the renderer. The warning message if needed can be added to `p5.Renderer` base class but I'm not entirely sure if it is useful.
> 
> The feature is also not complete (to be completed in #7270), this PR just to lay groundwork for converting the WebGL module to use the new module syntax.

Yes thanks, makes a lot of sense.